v2/selective.go

```diff
@@ -0,0 +1,268 @@
+package car
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"os"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipld/go-car/v2/index"
+	"github.com/ipld/go-car/v2/internal/carv1"
+	"github.com/ipld/go-car/v2/internal/loader"
+	ipld "github.com/ipld/go-ipld-prime"
+	"github.com/ipld/go-ipld-prime/linking"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	"github.com/ipld/go-ipld-prime/node/basicnode"
+	"github.com/ipld/go-ipld-prime/traversal"
+	"github.com/ipld/go-ipld-prime/traversal/selector"
+)
+
+// ErrSizeMismatch is returned when a written traversal realizes the written header size does not
+// match the actual number of car bytes written.
+var ErrSizeMismatch = fmt.Errorf("car-error-sizemismatch")
+
+// ErrOffsetImpossible is returned when specified paddings or offsets of either a wrapped carv1
+// or index cannot be satisfied based on the data being written.
+var ErrOffsetImpossible = fmt.Errorf("car-error-offsetimpossible")
```

Comment:
* Maybe `Invalid` or `Illegal` instead of `Impossible`? They tend to be the terms more commonly used to signal such issues.
* Human readable message 
* `errors.New` instead of `fmt.Errorf` since there isn't anything to format